metaflow/plugins/kfp/tests/flows/kfp_step_flow.py

```diff
@@ -0,0 +1,51 @@
+from metaflow import FlowSpec, step, kfp
+import os
+
+from kubernetes import client, config
+
+
+def assert_step_image(kfp_step_image: str):
+    is_on_kubernetes = os.getenv("K8S_CLUSTER_NAME")
+    if is_on_kubernetes:  # only perform this test on the cluster, not on local machine
+        config.load_incluster_config()
+        core_api_instance = client.CoreV1Api()
+
+        current_pod_name = os.environ.get("HOSTNAME", None)
+        current_pod_namespace = os.environ.get("POD_NAMESPACE", None)
+
+        if current_pod_name and current_pod_namespace:
+            pod_detail = core_api_instance.read_namespaced_pod(
+                namespace=current_pod_namespace, name=current_pod_name
+            )
+
+            for container_status in pod_detail.status.container_statuses:
+                if container_status.name == "main":
+                    assert container_status.image == kfp_step_image
+
+
+class KfpStepFlow(FlowSpec):
```

Comment:
We already `KfpFlow` that we can reuse and put this decorator on the start step.

https://github.com/zillow/metaflow/blob/tz/datasets/metaflow/plugins/kfp/tests/flows/kfp_flow.py#L16-L16